src/state.rs

```diff
@@ -0,0 +1,80 @@
+use std::time::{Duration, Instant};
+use crate::road::Road;
+use crate::pedestrian::Pedestrian;
+
+trait Vehicle {
+
+}
+
+trait State {
+
+    // const ROAD_LENGTH: f32;
+    // const CROSSINGS: Vec<Crossing>;
+
+
+    // fn get_vehicles(&self) -> &[dyn Vehicle];
+    fn timestamp(&self) -> Instant;
+
+    // get the road
+    fn get_road(&self) -> Road;
+
+    // get the list of vehicles
+    fn get_vehicles(&self) -> Vec<Box<dyn Vehicle>>;
```

Comment:
This will return an owned copy of the vector of `Vehicle` objects, which is probably not what we want. I would consider a slice `$[Box<dyn Vehicle>]` or, probably more appropriate, an iterator of some kind.